lib/ansible/modules/cloud/vmware/vmware_host_powermgmt_policy.py

```diff
@@ -0,0 +1,210 @@
+    """
+    Class to manage power management policy of an ESXi host system
+    """
+    def __init__(self, module):
+        super(VmwareHostPowerManagement, self).__init__(module)
+        cluster_name = self.params.get('cluster_name')
+        esxi_host_name = self.params.get('esxi_hostname')
+        self.hosts = self.get_all_host_objs(cluster_name=cluster_name, esxi_host_name=esxi_host_name)
+        if not self.hosts:
+            self.module.fail_json(msg="Failed to find host system with given configuration.")
+
+    def ensure(self):
+        """
+        Function to manage power management policy of an ESXi host system
+        """
+        results = dict(changed=False, result=dict())
+        policy = self.params.get('policy')
+        host_change_list = []
+        for host in self.hosts:
+            changed = False
+            results['result'][host.name] = dict(msg='', error='')
+
+            if policy == 'high-performance':
+                policy_key = 1
+                policy_short_name = "static"
+            elif policy == 'balanced':
+                policy_key = 2
+                policy_short_name = "dynamic"
+            elif policy == 'low-power':
+                policy_key = 3
+                policy_short_name = "low"
+            elif policy == 'custom':
+                policy_key = 4
+                policy_short_name = "custom"
+
+            power_system = host.configManager.powerSystem
+
+            # get current power policy
+            power_system_info = power_system.info
+            current_host_power_policy = power_system_info.currentPolicy
+            # current_host_power_policy.key
+            # current_host_power_policy.name
+            # current_host_power_policy.shortName
+            # current_host_power_policy.description
+            if current_host_power_policy.shortName == "static":
+                current_policy = 'high-performance'
+            elif current_host_power_policy.shortName == "dynamic":
+                current_policy = 'balanced'
+            elif current_host_power_policy.shortName == "low":
+                current_policy = 'low-power'
+            elif current_host_power_policy.shortName == "custom":
+                current_policy = 'custom'
+            if current_host_power_policy.key == policy_key:
+                results['result'][host.name]['msg'] = "power policy for host '%s' already configured" % host.name
+            else:
+                # get available power policies and check if policy is included
+                supported_policy = False
+                power_system_capability = power_system.capability
+                available_host_power_policies = power_system_capability.availablePolicy
+                for available_policy in available_host_power_policies:
+                    # available_policy.key
+                    # available_policy.name
+                    # available_policy.shortName
+                    # available_policy.description
```

Comment:
If these are not used, better to remove the commented variables

Reply:
ok